### PR TITLE
[add-s3-backup-support] 

### PR DESCRIPTION
**Pull request by aliartiza75**
PR will add support to back up the data in an AWS S3 bucket.

### CODE REVIEW

**Comment by naseemkullah** [comment]
There seems to be missing changes in the cronjob template @aliartiza75

**Comment by aliartiza75** [comment]
@naseemkullah yes i will add them,

**Comment by rawkode** [comment]
@aliartiza75 Would you like any assistance making those final changes, as requested by @naseemkullah ?

File: charts/influxdb/values.yaml [hunk 1]
```diff
@@ -235,3 +235,9 @@ backup:
   #   storageAccountSecret: influxdb-backup-azure-key
   #   destination_container: influxdb-container
   #   destination_path: ""
+
+  ## Amazon S3 Bucket
+  # awsS3:
+  #   s3userKey:
+  #   s3userKeySecret:
+  #   s3bucketName: influxdb-backup
```
Comment (naseemkullah) [inline charts/influxdb/values.yaml]:
```suggestion
  ## Amazon S3
  # amazonS3:
  #   userKey:
  #   userKeySecret:
  #   bucketName: influxdb-backup
```

nit: Amazon S3 seems to be the name of the service as opposed to AWS S3 as per [AWS documentation](https://aws.amazon.com/s3/)

Furthermore, the keys within the amazonS3 struct don't need to specify that they are S3 as that is implied.

Comment (aliartiza75) [reply charts/influxdb/values.yaml]: thank you @naseemkullah

Comment (aliartiza75) [reply charts/influxdb/values.yaml]: @naseemkullah are these the changes that you wanted, or do you have anything else in your mind?

**Comment by aliartiza75** [comment]
hi @rawkode, @naseemkullah  has already added those changes.

**Comment by naseemkullah** [comment]
Hi @aliartiza75, actually I only suggested some changes for the value names, you can commit them now after reviewing.

After settling on those value names, we still need to add the logic in the [backup cronjob template](https://github.com/influxdata/helm-charts/blob/17a922a33428878605470d95c95b7f4063dec3be/charts/influxdb/templates/backup-cronjob.yaml).

File: charts/influxdb/values.yaml (hunk 1 above)
Comment (naseemkullah) [reply charts/influxdb/values.yaml]: Please commit the suggestion if you do agree with those value name changes. This only covers the value names. As per my other comment we still need to add the logic in the backup cronjob template.

**Comment by julian-r** [comment]
@aliartiza75 anything i could help you with to get some traction here?

**Comment by aliartiza75** [comment]
@julian-r yes i will do it asap. I was a bit busy with some other tasks.

**Comment by cwash** [comment]
Hey folks just checking if there is anything I can do to help with this?

**Comment by naseemkullah** [comment]
Hi @cwash @julian-r , due to the time this is taking, I think a duplicate PR would be appropriate and we could merge the one that is finalized first.

@aliartiza75 please let us know if you plan on working on this, if not we can close this PR and let someone else pick up the issue, cheers.

**Comment by rawkode** [comment]
Closing, please re-open if you find the time to finish this. It would be a very welcomed PR :)

**Comment by trojanowski** [comment]
I created an alternative PR in #123.

**Pull request closed without merging**